src/components/AddPresetAsset.tsx

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import {
+  Button,
+  Checkbox,
+  Heading2,
+  InfoBlock,
+  InfoBlockVariant,
+  Loader,
+} from "@stellar/design-system";
+import { TextLink } from "components/TextLink";
+import { presetAsset } from "constants/presetAssets";
+import { getErrorMessage } from "helpers/getErrorMessage";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import { getPresetAssets } from "helpers/getPresetAssets";
+import { getValidatedUntrustedAsset } from "helpers/getValidatedUntrustedAsset";
+import { log } from "helpers/log";
+import { searchParam } from "helpers/searchParam";
+import { useRedux } from "hooks/useRedux";
+import { ActionStatus, SearchParams } from "types/types.d";
+import { shortenStellarKey } from "helpers/shortenStellarKey";
+
+export const AddPresetAsset = ({ onClose }: { onClose: () => void }) => {
+  const { account, allAssets, settings, untrustedAssets } = useRedux(
+    "account",
+    "allAssets",
+    "settings",
+    "untrustedAssets",
+  );
+  const [presetAssets, setPresetAssets] = useState<presetAsset[]>([]);
+  const [checkedAssets, setCheckedAssets] = useState<{
+    [key: string]: boolean;
+  }>({});
+  const [isValidating, setIsValidating] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const history = useHistory();
+
+  useEffect(() => {
+    setPresetAssets(getPresetAssets(allAssets.data));
+  }, [allAssets]);
+
+  useEffect(() => {
+    if (untrustedAssets.status === ActionStatus.SUCCESS) {
+      onClose();
+    }
+
+    if (untrustedAssets.errorString) {
+      setErrorMessage(untrustedAssets.errorString);
+    }
+  }, [onClose, untrustedAssets.errorString, untrustedAssets.status]);
+
+  const getAssetId = (asset: presetAsset) =>
+    `${asset.assetCode}:${asset.homeDomain || asset.issuerPublicKey}`;
+
+  const handleOnChangeCheckbox = (asset: presetAsset) => {
+    const updatedCheckedAssets = JSON.parse(JSON.stringify(checkedAssets));
```

Comment:
I may be wrong, but since JSON.parse and JSON.stringify are inverses of each other, wouldn't `JSON.parse(JSON.stringify(checkedAssets))` be the same as just `checkedAssets`?